src/models/BibDetails.ts

```diff
@@ -0,0 +1,164 @@
+import type { ProcessedSearchResult, SearchResult } from "../types/searchTypes"
+import type { LinkedBibDetail, BibDetail } from "../types/bibDetail"
+import { preProcess } from "../utils/bibModelPreprocessing"
+
+export default class BibDetailsModel {
+  bib: ProcessedSearchResult
+
+  constructor(bib: SearchResult) {
+    // Preprocessing includes grouping notes into an object with labels as keys,
+    // as well as interleaving parallels into primary fields
+    this.bib = preProcess(bib)
+    // this.subjectLiteral = "to do investigate  subject heading api fallback scenario"
+  }
+
+  buildHoldingDetail(fieldMapping: { label: string; field: string }) {
+    const bibFieldValue = this.bib.holdings[fieldMapping.field]
+    return this.buildDetail(fieldMapping.label, bibFieldValue)
+  }
+
+  buildStandardDetail(fieldMapping: { label: string; field: string }) {
+    const bibFieldValue = this.bib[fieldMapping.field]
+    return this.buildDetail(fieldMapping.label, bibFieldValue)
+  }
+
+  buildDetail(label, value) {
+    if (!value?.length) return null
+    return { label, value }
+  }
+
+  buildInternalLinkedDetail(fieldMapping: {
+    label: string
+    field: string
+  }): LinkedBibDetail {
+    const value = this.bib[fieldMapping.field]
+    if (!value?.length) return null
+    return {
+      link: "internal",
+      label: fieldMapping.label,
+      value: value.map((v: string) => {
+        const internalUrl = `/search?filters[${fieldMapping.field}][0]=${v}`
+        return { url: internalUrl, urlLabel: v }
+      }),
+    }
+  }
+
+  get extent(): BibDetail {
```

Comment:
What's "extent" in regards to a bib?

Reply:
I don't know how to verify this business logic but I know this code is mostly from DFE. Is there a doc I can read up on dimensions and extents?

Reply:
extent is like how many pages a book is, and dimensions are the size of the book, like "30cm x 40cm". I don't think this is documented anywhere besides the ticket for this work. I can probably dig it up for you if you want to see it.

Reply:
This is mostly the number of pages with some other details. I believe it's coming from the MARC 300 field, and what gets stored here as`extent` and `dimensions` are the $a and $b subfields. All the `removeSemiColon` business is because the subfields are cataloged with punctuation (semicolons) so that they can just be joined together, but we only show $a and $b and there is extra trailing punctuation if the record has $c etc and we don't remove it